File: BAEKJOON/Py/DS/10773.py
#O(n) stack

k = int(input())

arr = []

for _ in range(k):
    n = int(input())
    if n == 0:
        arr.pop()
    else:
        arr.append(n)


print(sum(arr))

#GPT
# import sys
# stack = []
# k = int(sys.stdin.readline())
# for _ in range(k):
#     num = int(sys.stdin.readline())
#     if num != 0:
#         stack.append(num)
#     else:
#         stack.pop()
# print(sum(stack))

File: BAEKJOON/C++/DS/10773.cpp
#include <iostream>
#include <stack>

long long sum_stack_destructive(std::stack<int>& st) {
    long long sum = 0;
    while (!st.empty()) {
        sum += st.top();
        st.pop();
    }
    return sum;
}

int main(){
    int K, num;    
    std::cin >> K;
    std::stack<int> st;
    
    for (int i = 0; i < K; i++){

        std::cin >> num;
        if(num == 0){
            st.pop();
        }else{
            st.push(num);
        }
    }
    std::cout << sum_stack_destructive(st);
}